src/client.js

```diff
@@ -305,427 +305,574 @@ class ConnectionParameters {
+      await client.NLWS.xtkSession.write(fileRes);
+    }
 
-// ========================================================================================
-// ACC Client
-// ========================================================================================
+    /**
+     * This will call the SOAP method(PublishIfNeeded) to publish the fileRes object
+     * @param fileRes
+     * @returns {Promise<void>}
+     * @private
+     */
+    const _publishIfNeeded = async (fileRes) => {
+      await client.NLWS.xtkFileRes.create(fileRes).publishIfNeeded();
+    }
 
-/**
+    /**
+     * This Will call the SOAP method(GetURL) which returns the public URL of the uploaded file.
+     * @param fileRes
+     * @returns {Promise<string>}
+     * @private
+     */
+    const _getPublicUrl = async (fileRes) => {
+      return await client.NLWS.xtkFileRes.create(fileRes).getURL()
+    }
+    return {
+      /**
+       * This is the exposed/public method for fileUploader instance which will do all the processing related to the upload process internally and returns the promise containing all the required data.
+       * @param file, where file is an instance of [File](https://developer.mozilla.org/en-US/docs/Web/API/File)
+       * @returns {Promise<{name: string, md5: string, type: string, size: string, url: string}>}
+       */
+      upload: (file) => {
+        return new Promise((resolve, reject) => {
+          if (!Util.isBrowser()) {
+            reject('File uploading is only supported in browser based calls.');
+          }
+          try {
+            const data = new FormData()
+            data.append('file_noMd5', file)
+            //TODO: Needs to be refactored after cookie issue get resolved.
+            client._makeHttpCall({
+              url: `${client._connectionParameters._endpoint}/nl/jsp/uploadFile.jsp`,
+              processData: false,
+              credentials: 'include',
+              method: 'POST',
+              body: data,
+              headers: {
+                'x-security-token': client._securityToken,
+                'Cookie': '__sessiontoken=' + client._sessionToken,
+              }
+            }).then((okay)=>{
+              if(!okay.startsWith('Ok')){
+                throw okay;
+              }
+              const iframe = document.createElement('iframe');
+              iframe.style.height = 0;
+              iframe.style.width = 0;
+              document.controller = { // Written to  support https://git.corp.adobe.com/Campaign/ac/blob/v6-master/nl/datakit/nl/jsp/uploadFile.jsp
+                uploadFileCallBack: async (data) => {
+                  if(!data || data.length !== 1){
+                    // Tried to replicate the logic for file upload functionality written here:
+                    // https://git.corp.adobe.com/Campaign/ac/blob/v6-master/wpp/xtk/web/dce/uploader.js
+                    reject(CampaignException.FILE_UPLOAD_FAILED('MALFORMED DATA'+ data.toString()));
```

Comment:
This is not proper english. Logging potentially very long variables (data) should be avoided